ui/src/app/store/general/types.ts

```diff
@@ -75,9 +75,10 @@ export type MachineActionName =
 
 export type MachineAction = {
   name: MachineActionName;
-  sentence: string;
-  title: string;
-  type: string;
+  sentence?: string;
+  title?: string;
+  type?: string;
+  hardwareType?: number;
```

Comment:
I don't think this is the right approach. `MachineAction` here is returned from the server and will always have the top 4 properties. `hardwareType` is specific only to how it's being used in machine details, so should probably be a different type that lives there.